lib/stringtheory/src/interning/map.rs

```diff
@@ -253,15 +254,16 @@ impl InternerStorage {
 
         Self {
             ptr,
-            len: 0,
+            offset: 0,
             capacity,
+            len: 0,
             reclaimed: ReclaimedEntries::new(),
         }
     }
 
     /// Returns the total number of unused bytes that are available for interning.
     fn available(&self) -> usize {
-        self.capacity.get() - self.len
+        self.capacity.get() - self.offset
```

Comment:
So this is actually where we _would_ want to use `self.len`, because we're trying to make `self.len` indicate how many bytes are _actually_ in use, not just how far into the buffer we've managed to go.